tutorials/09-set-up-physics-engine-tpe.md

```diff
@@ -0,0 +1,246 @@
+  const Identity &, const std::string &_name)
+{
+  auto world = std::make_shared<tpelib::World>();
+  world->SetName(_name);
+  return this->AddWorld(world);
+}
+
+... // other function implementations
+```
+
+To realize the \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+on the TPE physics engine, we implement `ConstructEmptyWorld` method. We simply
+instantiate \ref ignition::physics::tpelib::World "World" object and then
+wrap it with `std::share_ptr`. We then set the corresponding input world name
+and store the current live world object.
+
+Please take a look at [EntityManagement_TEST.cc](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/EntityManagement_TEST.cc)
+as an example unit test for this `EntityManagementFeatures` "FeatureList".
+
+### Main plugin.cc file
+
+After the required \ref ignition::physics::FeatureList "FeatureList" are implemented,
+we will include all the "FeatureLists" in [plugin.cc](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/plugin.cc)
+main file and register the TPE physics plugin as follow:
+
+```cpp
+#include <ignition/physics/Register.hh>
+
+#include "Base.hh"
+
+#include "CustomFeatures.hh"
+#include "EntityManagementFeatures.hh"
+#include "FreeGroupFeatures.hh"
+#include "KinematicsFeatures.hh"
+#include "SDFFeatures.hh"
+#include "ShapeFeatures.hh"
+#include "SimulationFeatures.hh"
+
+namespace ignition {
+namespace physics {
+namespace tpeplugin {
+
+struct TpePluginFeatures : FeatureList<
+  CustomFeatureList,
+  EntityManagementFeatureList,
+  FreeGroupFeatureList,
+  KinematicsFeatureList,
+  SDFFeatureList,
+  ShapeFeatureList,
+  SimulationFeatureList
+> { };
+
+class Plugin :
+  public virtual Implements3d<TpePluginFeatures>,
+  public virtual Base,
+  public virtual CustomFeatures,
+  public virtual EntityManagementFeatures,
+  public virtual FreeGroupFeatures,
+  public virtual KinematicsFeatures,
+  public virtual SDFFeatures,
+  public virtual ShapeFeatures,
+  public virtual SimulationFeatures { };
+
+IGN_PHYSICS_ADD_PLUGIN(Plugin, FeaturePolicy3d, TpePluginFeatures)
+
+}
+}
+}
+```
+
+In general, there are 3 steps for `plugin.cc`:
+- Define the final \ref ignition::physics::FeatureList "FeatureList" including
+all required "FeatureLists". In TPE case, it is `TpePluginFeatures`.
+- Define an empty class inherited all "FeatureLists" class, [Base](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/Base.hh)
+class and \ref ignition::physics::Implements "Implements" class implementing
+\ref ignition::physics::FeaturePolicy "FeaturePolicy" 2D or 3D and different
+scalar type.
+- Register the physics plugin using `IGN_PHYSICS_ADD_PLUGIN` macro (See
+  \ref createphysicsplugin "Implement a physics plugin" for more detail).
+
+An example of `CMakeLists.txt` for building the TPE plugin can be found [here](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/CMakeLists.txt).
```

Comment:
I think it'd be nice to again point out at this point what files are in this folder, so that users can compile locally following this tutorial. Ideally it looks something like this tutorial https://index.ros.org/doc/ros2/Tutorials/Writing-A-Simple-Cpp-Publisher-And-Subscriber/  where users are able to compile the all the example code shown in the tutorial with clear instruction.

Reply:
This `CMakeLists.txt` file is too context-dependent, it requires tags set in upper `CMakeLists.txt` level. I think we should not explain `CMakeLists.txt` here but let developer to download it and compile the new plugin. I will mention the folder structure